### PR TITLE
console: converting routes.js and server.js to typescript fix #4314

### PR DESCRIPTION
**Pull request by sudeshna-22**
### Changelog

- [ ] `CHANGELOG.md` is updated with user-facing content relevant to this PR. If no changelog is required, then add the `no-changelog-required` label.

### Affected components

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System
- [ ] Tests
- [ ] Other (list it)

#### Catalog upgrade

Does this PR change Hasura Catalog version?
- [ ] No
- [ ] Yes
  - [ ] Updated docs with SQL for downgrading the catalog 

#### Metadata

Does this PR add a new Metadata feature?
- [ ] No
- [ ] Yes
  - Does `run_sql` auto manages the new metadata through schema diffing?
    - [ ] Yes
    - [ ] Not required
  - Does `run_sql` auto manages the definitions of metadata on renaming?
    - [ ] Yes
    - [ ] Not required
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?
    - [ ] Yes
    - [ ] Not required

#### GraphQL
- [ ] No new GraphQL schema is generated
- [ ] New GraphQL schema is being generated:
   - [ ] New types and typenames are correlated
   
   

#### Breaking changes

- [ ] No Breaking changes
- [ ] There are breaking changes:

  1. Metadata API

     Existing `query` types:
     - [ ] Modify `args` payload which is not backward compatible
     - [ ] Behavioural change of the API
     - [ ] Change in response `JSON` schema
     - [ ] Change in error code
     

  2. GraphQL API

     Schema Generation:
     
     
     - [ ] Change in any `NamedType`
     - [ ] Change in table field names
     

     Schema Resolve:-
     
     - [ ] Change in treatment of `null` value for any input fields 
     

  3. Logging

     - [ ] Log `JSON` schema has changed
     - [ ] Log `type` names have changed

### CODE REVIEW

File: console/package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
   "keywords": [],
   "scripts": {
     "start-dev": "node -r dotenv/config ./bin/server.js",
-    "watch-client": "TAILWIND_MODE=watch node -r dotenv/config webpack/webpack-dev-server.js",
```
Comment (bipsBro) [inline console/package.json:18]: No need to commit this change. revert this change

File: console/package.json [hunk 2]
```diff
@@ -72,6 +72,7 @@
     "apollo-link-ws": "1.0.20",
     "axios": "0.27.2",
     "brace": "0.11.1",
+    "caniuse-lite": "^1.0.30001358",
```
Comment (bipsBro) [inline console/package.json:75]: any reason for installing this package?

Comment (sudeshna-22) [reply console/package.json:75]: It was not possible to run in the browser without updating this package. It recurrently displayed the message "the package is out of date."

File: console/package.json (hunk 1 above)
Comment (sudeshna-22) [reply console/package.json:18]: Sure
